rfcs/0000-observability-api.md

```diff
@@ -0,0 +1,148 @@
+| Tracing               | **endpoint** , **tags**  , **insecure**, **strageyRules** |
+| Metrics               | **enableService**, **deep**                               |
+| Alerts *              | **namespace**, **enable**                                 |
+| Dashboards *          | **namespace**, **enable**                                 |
+| Other 3rd Party *     | e.g Kiali  **enable** bool                                |
+
+###### **Note**: Observability pieces with a * denotes these are post v1 milestone
+            
+
+### Sample use case (With whats currently available)
+A use case a user might have could be the following:
+
+1. Setup tracing for the Kuadrant components Authorino and Limitador with the required endpoint and optional tag. 
+2. Disable the creation of the service and serviceMonitors created by Kuadrant for the components and there operators
+3. Configure logging for Authorino and Limitador. Note: For the first phase this is not available for the operators as they dont use the component CRs to implement new log levels flags/env-vars are used instead.
+
+```yaml 
+
+apiVersion: Kuadrant.io/v1alpha1
+kind: kuadrant
+spec:
+  observability:
+    tracing: 
+        endpoint: rpc://tempo.tempo.svc.cluster.local:4317
+        tags: tag1, tag2
+    metrics: 
+        enable: false
+    logging:
+      logLevel: debug      
+```
+
+### Status
+The status of the Kuadrant CR will not be the observability stack is in a "healthy" state i.e Prometheus and Grafana is up and running. It should be the status of only the things that we have control over. The status will show what components the user will have configured for example logging setup for Authorino and Limitador. We will not be taking responsibility for aspects we don't have control over. 
+
+#### Example status:
+
+```yaml 
+logging:
+  authorino: debug
+  limitador: debug
+tracing:
+  authorino: endpoint, tags
+  limitador: endpoint, tags
+metrics:
+  disabled
+```
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+In terms of how the information supplied in the Kuadrant CR will get passed to the other Kuadrant components, multiple options have been brought up:
+
+- directly - like setting flags or configuration directly on the deployments of the different Kuadrant components.
+- indirectly -  Passing the information to Authorino & Limitador via the Authorino & Limitador CRs
+- indirectly -  Passing the information to Authorino & Limitador in the form of there own Observability CRs
+
+The best approach would be the indirect approach, meaning once the Kuadrant CR is updated, the information is passed to the relevant component CR. For example the tracing section in the Authorino CR spec would be updated with the required endpoints and other configuration from the Kuadrant CR. This does mean that the spec exposed is whats in the Kuadrant component CR's but new changes can be requested and implemented in said components.
```

Comment:
```suggestion
1. directly - like setting flags or configuration directly on the deployments of the different Kuadrant components.
2. indirectly -  Passing the information to Authorino & Limitador via the Authorino & Limitador CRs
3.  indirectly -  Passing the information to Authorino & Limitador in the form of there own Observability CRs

The best approach would be the approach 2), an indirect approach, meaning once the Kuadrant CR is updated, the information is passed to the relevant component CR. For example, the tracing section in the Authorino CR spec would be updated with the required endpoints and other configuration from the Kuadrant CR. This does mean that the spec exposed is whats in the Kuadrant component CR's, but new changes can be requested and implemented in said components.
```

Maybe use a numbered list instead since the description seems to describe the first indirect approach specificially? 🤔